Dockerfile.cuda11.8

```diff
@@ -0,0 +1,43 @@
+FROM nvidia/cuda:11.8.0-runtime-ubuntu22.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update
+RUN apt-get install -y software-properties-common
+RUN add-apt-repository -y ppa:deadsnakes/ppa
+RUN apt-get update
+RUN apt-get install -y python3.11 python3.11-venv python3.11-dev python3-pip git cuda-nvcc-11-8 cuda-toolkit-11-8 vim
+RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.11 1
+RUN python3.11 -m ensurepip --upgrade
```

Comment:
`python3-pip` has already been installed, we don't need to double check it I suppose.

Reply:
regarding pip:
its necessary, from the apt repo provides pip v22.0.2, therefore we need to upgrade it with pip itself after installed.
> python3 -m ensurepip --upgrade
will upgrade pip to v24.0 which then is necessarry for further steps (i cant remember excactly, but i remember somehow it was connected to ensure setuptools and wheel can be installed successfully)